### ISSUE TITLE
Use Asset identifier instead of id in realtive URL

### ISSUE THREAD

**Issue by jorgenjanssonlee** [robhabraken/content-hub-focal-point-cropping#16]
Hi Rob,

When using the Asset ID in the realtiveURL for the public link, it becomes very easy for external parties to work out the URL pattern and simply iterate through incrementing integers with a given ratio/name to get all public links from a content hub instance. 
In some cases, where public links are used for staging/testing environments and not yet ready to go live in prod, this becomes a security issue.

I think it's better to use the Asset Identifier as it's a longer, and random, string of characters. It is still a unique identifier within the instance, so shouldn't change any other functionality in the cropping script, e.g. retrieving existing public links for updates etc.

I've made the following changes for one deployment and it passes all our tests and seem to work well. I also added some more logging...

Would you consider implementing this in the default script?

- comment out `var title = asset.GetPropertyValue("Title").ToString();` and replace it with `var title = asset.Identifier;`

- Add  additional logging
   `MClient.Logger.Info($"Starting script for {assetId}");`
    `MClient.Logger.Info($"Identifier is: {title}");`

![image](https://user-images.githubusercontent.com/28835885/227131029-11a87aa2-b164-4612-860d-cdb7762d1c48.png)

**Comment by robhabraken** [comment]
Hi Jorgen,

thanks for the good suggestion. As I indeed shared earlier, we do use the asset ID and are okay with the potential exposure since only approved assets will be published. But I like your suggestion and will test it out as the default for my script instead of the title, because it is less variable than the title and also more secure.

**Comment by robhabraken** [comment]
Replying to your code example: I think it is better to supply the value in line 161 as input for creating the cropping definition object, instead of overriding the title attribute, as you assign an Identifier value to a title, but it isn't the title. I will mock up a version myself and test it in my sandbox before creating a pull request for it.

**Comment by robhabraken** [comment]
Please see https://github.com/robhabraken/content-hub-focal-point-cropping/pull/19 for the suggested implementation.

**Comment by jorgenjanssonlee** [comment]
Hi Rob, I definitely like your implementation better and have deployed that now. Works well in our testing so feel free to close the issue once you've merged the code.

I was adding the extra logging because in earlier testing the script threw an error (rendition/mainfile race condition) before anything had been logged. the error didn't mention which asset it was working on, so I added the logging as early as possible to help troublshooting.
I've now reworked it to be a single line, see below.
I will keep it like this in our script as it's been useful a couple of time, but you certainly don't need to add it to the source script :)
![image](https://user-images.githubusercontent.com/28835885/227394301-5ca0b8c9-0911-4223-8878-79a5200d2673.png)
